### PR TITLE
Update attribute arguments table

### PR DESCRIPTION
**Pull request by keerlu**
## Describe this PR

Update of the schema reference to go with #2514. This is an updated version of #2532, which I will close.

### CODE REVIEW

File: content/400-reference/200-api-reference/100-prisma-schema-reference.mdx [hunk 1]
```diff
@@ -1040,7 +1040,16 @@ Defines a single-field ID on the model.
...
+| Name  | Required | Type     | Description |
+| ----- | -------- | -------- | ----------- |
+| `map` | **No**   | `String` |             |
+
+**Note:** In version 3.5.0 and later, new `length` and `sort` arguments are available as a Preview feature. See [Index configuration](../../../../concepts/components/prisma-schema/indexes) for more details.
+
+| Name     | Required | Type     | Description                                                                                                                                                             |
+| -------- | -------- | -------- | ----------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `length` | **No**   | `number` | **MySQL only.** Allows you to specify a maximum length for the subpart of the value to be indexed.                                                                      |
+| `sort`   | **No**   | `String` | **SQL Server only.** Allows you to specify in what order the entries of the index or constraint are stored in the database. The available options are `asc` and `desc`. |
```
Comment (do4gr) [inline content/400-reference/200-api-reference/100-prisma-schema-reference.mdx]: `Asc` and `Desc` are the options, so first letter capitalized

File: content/400-reference/200-api-reference/100-prisma-schema-reference.mdx [hunk 2]
```diff
@@ -1319,6 +1336,13 @@ The name of the `fields` argument on the `@@id` attribute can be omitted:
 @@id([title, author])
 ```
 
+**Note:** In version 3.5.0 and later, new `length` and `sort` arguments are available as a Preview feature. See [Index configuration](../../../../concepts/components/prisma-schema/indexes) for more details.
+
+| Name     | Required | Type     | Description                                                                                                                                                             |
+| -------- | -------- | -------- | ----------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `length` | **No**   | `number` | **MySQL only.** Allows you to specify a maximum length for the subpart of the value to be indexed.                                                                      |
+| `sort`   | **No**   | `String` | **SQL Server only.** Allows you to specify in what order the entries of the index or constraint are stored in the database. The available options are `asc` and `desc`. |
```
Comment (do4gr) [inline content/400-reference/200-api-reference/100-prisma-schema-reference.mdx]: same as above.

File: content/400-reference/200-api-reference/100-prisma-schema-reference.mdx [hunk 3]
```diff
@@ -1522,6 +1548,7 @@ Defines a [default value for a field](/concepts/components/prisma-schema/data-mo
 | Name         | Required | Type                                      | Description |
 | ------------ | -------- | ----------------------------------------- | ----------- |
 | `expression` | **Yes**  | An expression (e.g. `5`, `true`, `now()`) |             |
+| `map`        | **No**   | String                                    |             |
```
Comment (do4gr) [inline content/400-reference/200-api-reference/100-prisma-schema-reference.mdx]: map on defaults is only available on SqlServer

File: content/400-reference/200-api-reference/100-prisma-schema-reference.mdx [hunk 4]
```diff
@@ -1819,20 +1846,35 @@ Defines a unique constraint for this field.
...
+| Name  | Required | Type     | Description |
+| ----- | -------- | -------- | ----------- |
+| `map` | **No**   | `String` |             |
+
+**Note:** In version 3.5.0 and later, new `length` and `sort` arguments are available as a Preview feature. See [Index configuration](../../../../concepts/components/prisma-schema/indexes) for more details.
+
+| Name     | Required | Type     | Description                                                                                                                                        |
+| -------- | -------- | -------- | -------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `length` | **No**   | `number` | **MySQL only.** Allows you to specify a maximum length for the subpart of the value to be indexed.                                                 |
+| `sort`   | **No**   | `String` | Allows you to specify in what order the entries of the index or constraint are stored in the database. The available options are `asc` and `desc`. |
```
Comment (do4gr) [inline content/400-reference/200-api-reference/100-prisma-schema-reference.mdx]: again Asc, Desc

File: content/400-reference/200-api-reference/100-prisma-schema-reference.mdx [hunk 5]
```diff
@@ -1819,20 +1846,35 @@ Defines a unique constraint for this field.
...
+> **Note**: Until version 3.0.0, the signature was:
 >
 > ```no-lines
 > @unique
 > ```
 
+> **Note**: In version 3.5.0 and later, the signature for the `"extendedIndexes"` Preview feature is:
+>
+> ```prisma no-lines
+> @unique(_ value: Expression, map: String?, length: number?, sort: String?)
```
Comment (do4gr) [inline content/400-reference/200-api-reference/100-prisma-schema-reference.mdx]: no _ value: Expression, probably a copy leftover from default

File: content/400-reference/200-api-reference/100-prisma-schema-reference.mdx [hunk 6]
```diff
@@ -2013,18 +2056,31 @@ The name of the `fields` argument on the `@@unique` attribute can be omitted:
 @@unique(fields: [title, author], name: "titleAuthor")
 ```
```
Comment (do4gr) [inline content/400-reference/200-api-reference/100-prisma-schema-reference.mdx:2058]: We should mention here that the length and sort args go as args into the field reference and are not their own top level arguments on compounds.

File: content/400-reference/200-api-reference/100-prisma-schema-reference.mdx [hunk 7]
```diff
@@ -2013,18 +2056,31 @@ The name of the `fields` argument on the `@@unique` attribute can be omitted:
...
 
+**Note:** In version 3.5.0 and later, new `length` and `sort` arguments are available as a Preview feature. See [Index configuration](../../../../concepts/components/prisma-schema/indexes) for more details.
+
+| Name     | Required | Type     | Description                                                                                                                                        |
+| -------- | -------- | -------- | -------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `length` | **No**   | `number` | **MySQL only.** Allows you to specify a maximum length for the subpart of the value to be indexed.                                                 |
+| `sort`   | **No**   | `String` | Allows you to specify in what order the entries of the index or constraint are stored in the database. The available options are `asc` and `desc`. |
```
Comment (do4gr) [inline content/400-reference/200-api-reference/100-prisma-schema-reference.mdx]: Asc, Desc

File: content/400-reference/200-api-reference/100-prisma-schema-reference.mdx [hunk 8]
```diff
@@ -2239,7 +2295,7 @@ Introspection and Prisma Migrate do not currently support MongoDB, which means P
 | Name     | Required | Type               | Description                                                                                                                                                                                                                                 |
 | -------- | -------- | ------------------ | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
 | `fields` | **Yes**  | `FieldReference[]` | A list of field names - for example, `["firstname", "lastname"]`                                                                                                                                                                            |
-| `name`   | **No**   | `name`             | The name of the index in the underlying database (Prisma generates an index name that respects identifier length limits if you do not specify a name. Prisma uses the following naming convention: `tablename.field1_field2_field3_unique`) |
```
Comment (do4gr) [inline content/400-reference/200-api-reference/100-prisma-schema-reference.mdx:2242]: there is still a name argument, but it refers to the name in the client api, see @@id

File: content/400-reference/200-api-reference/100-prisma-schema-reference.mdx [hunk 9]
```diff
@@ -2248,19 +2304,26 @@ The _name_ of the `fields` argument on the `@@index` attribute can be omitted:
 @@index([title, author])
 ```
```
Comment (do4gr) [inline content/400-reference/200-api-reference/100-prisma-schema-reference.mdx:2314]: same as for @@unique, these go into the field reference

File: content/400-reference/200-api-reference/100-prisma-schema-reference.mdx (hunk 6 above)
Comment (keerlu) [reply content/400-reference/200-api-reference/100-prisma-schema-reference.mdx:2058]: I've added an example snippet to demonstrate this - is this correct?

File: content/400-reference/200-api-reference/100-prisma-schema-reference.mdx (hunk 9 above)
Comment (keerlu) [reply content/400-reference/200-api-reference/100-prisma-schema-reference.mdx:2314]: I've added an example snippet to demonstrate this - is this correct?

Comment (do4gr) [reply content/400-reference/200-api-reference/100-prisma-schema-reference.mdx:2314]: yes.

**Pull request merged**